test/model/ai/evaluation/BoardEvaluatorTest.java

```diff
@@ -0,0 +1,82 @@
+package model.ai.evaluation;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import model.Board;
+import model.PieceColor;
+import model.ai.evaluation.BoardEvaluatorInterface;
+import model.ai.evaluation.KingCountEvaluator;
+import model.ai.evaluation.PawnCountEvaluator;
+import model.ai.evaluation.PlainBoardEvaluator;
+
+public class BoardEvaluatorTest {
+
+    @Test
+    public void testEvaluateBoard_KingCount() {
+        Board board = new Board();
+        board.getPiece(1).kingMe();
+        board.getPiece(2).kingMe();
+
+        BoardEvaluatorInterface boardEvaluator = new KingCountEvaluator(new PlainBoardEvaluator());
+
+        double epsilon = 0.0;
+
+        double expectedValue = 2.0;
+        double actualValue = boardEvaluator.evaluateBoard(board, PieceColor.BLACK);
+        assertEquals(expectedValue, actualValue, epsilon);
+
+    }
+
+    @Test
+    public void testEvaluateBoard_PawnAndKingCount() {
+        Board board = new Board(Arrays.asList(1, 2, 3, 4), Arrays.asList(30));
+        board.getPiece(1).kingMe();
+        board.getPiece(2).kingMe();
+
+        BoardEvaluatorInterface boardEvaluator = new KingCountEvaluator(
+                new PawnCountEvaluator(new PlainBoardEvaluator()));
+
```

Comment:
Ahh okay, cool! I think I have to call it a night since my Internet is
gone, but go ahead and merge it in for me, and I'll jump on plumbing it all
in tomorrow morning!

On Wed, Oct 14, 2015, 1:30 AM csperlich <notifications@github.com> wrote:

> In test/model/ai/evaluation/BoardEvaluatorTest.java
> <https://github.com/CSW-Checkers/checkers/pull/17#discussion_r41956253>:
>
> > +
> > +        double expectedValue = 2.0;
> > +        double actualValue = boardEvaluator.evaluateBoard(board, PieceColor.BLACK);
> > +        assertEquals(expectedValue, actualValue, epsilon);
> > +
> > +    }
> > +
> > +    @Test
> > +    public void testEvaluateBoard_PawnAndKingCount() {
> > +        Board board = new Board(Arrays.asList(1, 2, 3, 4), Arrays.asList(30));
> > +        board.getPiece(1).kingMe();
> > +        board.getPiece(2).kingMe();
> > +
> > +        BoardEvaluatorInterface boardEvaluator = new KingCountEvaluator(
> > +                new PawnCountEvaluator(new PlainBoardEvaluator()));
> > +
>
> There's another constructor that takes a weight, to modify the effect of
> the evaluator.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CSW-Checkers/checkers/pull/17/files#r41956253>.
>